test/rules/no-mocha-arrows.js

```diff
@@ -16,35 +16,73 @@ ruleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
             code: 'function it () {}; it(() => { console.log("okay") })',
             parserOptions: { ecmaVersion: 6 }
         },
-        'it.only()'
+        'it.only()',
+        'it(function(done) { assert(something, false); done(); })',
+        {
+            code: 'it(function*() { assert(something, false) })',
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: 'it(async function () { assert(something, false) })',
+            parserOptions: { ecmaVersion: 2017 }
```

Comment:
Pretty sure you can do
```js
ruleTester = new RuleTester({
  parserOptions: {ecmaVersion: 2017}
});
```
and remove all the parserOptions override in individual test cases.